.github/workflows/extension-tests.yml

```diff
@@ -0,0 +1,56 @@
+name: Extension Tests
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["**"]
+    paths:
+      - "extensions/**"
+      # Exclude RISC-V test vectors
+      - "!extensions/rv32im/**"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}
+  cancel-in-progress: true
+
+env:
+  CARGO_TERM_COLOR: always
+  OPENVM_FAST_TEST: "1"
+
+jobs:
+  extension-tests:
+    strategy:
+      matrix:
+        extension:
+          - {name: 'algebra', path: 'algebra'}
+          - {name: 'bigint', path: 'bigint'}
+          - {name: 'keccak256', path: 'keccak256'}
+          - {name: 'native', path: 'native'}
+          - {name: 'sha256', path: 'sha256'}
+      # Ensure tests run in parallel even if one fails
+      fail-fast: false
+    
+    runs-on:
+      - runs-on=${{ github.run_id }}
+      - runner=64cpu-linux-arm64
```

Comment:
can you add a `tag=extension-${{ matrix.extension.name }}` for unique runner assignment?